src/app.rs

```diff
@@ -525,6 +552,7 @@ pub fn build_app() -> Command<'static> {
                 .short('S')
                 .takes_value(true)
                 .number_of_values(1)
+                .value_parser(SizeFilter::from_string)
```

Comment:
Personally, I'd recommend splitting up the resolving of deprecations and using new features into separate PRs. 

This way it doesn't get blocked on discussions around items like

> On a first glance, the new code looks quite a bit more complicated than what we had before, or am I mistaken? The build process also seems to be more complicated with a new feature flag?